### PR TITLE
C# linq custom sequence example #164

### PR DESCRIPTION
**Pull request by matthewreily**
Added the linq custom sequence operator examples for #164 and parent task #90 cc @BethMassi @cartermp

### CODE REVIEW

File: samples/linq/csharp/customsequence/CustomSequence-Sample-1.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        //and sum the result.
+        //
+        //Output
+        // Dot product: 109
+        public static void Example()
+        {
+            int[] vectorA = {0, 2, 4, 5, 6};
+            int[] vectorB = {1, 3, 5, 7, 8};
+
+            int dotProduct = vectorA.Combine(vectorB, (a, b) => a*b).Sum();
```
Comment (cartermp) [inline samples/linq/csharp/customsequence/CustomSequence-Sample-1.cs]: Nitpick: should have whitespace surrounding the `*`: `(a, b) => a * b`.

File: samples/linq/csharp/customsequence/Program.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+namespace CustomSequence
+{
+    public class Program
+    {
+        //Entry point of application
+        public void Main(string[] args)
+        {
+            //Invoke the linq custom sequence execution examples within the project
+            CustomSequenceSample1.Example();
+
```
Comment (cartermp) [inline samples/linq/csharp/customsequence/Program.cs]: Nitpick: extraneous line break.

**Comment by cartermp** [comment]
LGTM modulo minor feedback!

**Comment by matthewreily** [comment]
made changes suggested by @cartermp and squashed commit

**Comment by cartermp** [comment]
Thanks!

**Pull request merged**